modules/slugs/includes/qtranslate-slug-import.php

```diff
@@ -0,0 +1,149 @@
+    if ( ! empty( $msg ) ) {
+        // Found some post/term meta with the new keys, no import to suggest (it can still be done manually).
+        return '';
+    }
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+
+    return empty ( $msg ) ? $msg : implode( '<br>', $msg );
+}
+
+/**
+ * Import slugs meta by duplicating the legacy QTS postmeta and termmeta for QTX.
+ *
+ * @param bool $db_commit true to commit changes, false for dry-run mode.
+ *
+ * @return string messages giving details.
+ */
+function qtranxf_slugs_import_qts_meta( $db_commit ) {
+    global $wpdb;
+
+    $new_prefix = QTX_SLUGS_META_PREFIX;
+    $old_prefix = QTX_SLUGS_LEGACY_QTS_META_PREFIX;
+
+    /**
+     * Generic function that imports old meta into new ones. All existing new meta are erased first.
+     *
+     * @param string $table name of the meta table (postmeta, termmeta)
+     * @param string $colid column name giving the meta id (post_id, term_id)
+     * @param string[] $msg array of messages, updated
+     *
+     * @return void
+     */
+    $import_meta = function ( $table, $colid, &$msg ) use ( $wpdb, $old_prefix, $new_prefix ) {
+        $results = $wpdb->query( "DELETE FROM $table WHERE meta_key like '$new_prefix%'" );
+        if ( $results ) {
+            $msg[] = sprintf( __( "Deleted %s rows from $table (%s).", 'qtranslate' ), $results, $new_prefix );
+        }
+        $results = $wpdb->query( "INSERT INTO $table ($colid, meta_key, meta_value)
+                              SELECT $colid, REPLACE(meta_key, '$old_prefix', '$new_prefix'), meta_value
+                              FROM  $table
+                              WHERE meta_key like '$old_prefix%'" );
+        $msg[]   = sprintf( __( "Imported %s rows into $table (%s->%s).", 'qtranslate' ), $results ?: '0', $old_prefix, $new_prefix );
+    };
+
+    $msg = [];
+    $wpdb->query( "START TRANSACTION" );
+    $import_meta( $wpdb->postmeta, 'post_id', $msg );
+    $import_meta( $wpdb->termmeta, 'term_id', $msg );
+    if ( $db_commit ) {
+        $wpdb->query( "COMMIT" );
+    } else {
+        $wpdb->query( "ROLLBACK" );
+    }
+
+    return implode( '<br>', $msg );
+}
+
+/**
+ * Import slugs legacy QTS options.
+ *
+ * @param bool $db_commit true to commit changes, false for dry-run mode.
+ *
+ * @return string messages giving details.
+ */
+function qtranxf_slugs_import_qts_options( $db_commit ) {
+    $msg = [];
+
+    $new_options = get_option( QTX_OPTIONS_MODULE_SLUGS );
+    if ( $new_options ) {
+        if ( $db_commit ) {
+            delete_option( QTX_OPTIONS_MODULE_SLUGS );
+        }
+        $msg[] = sprintf( __( "Deleted %s types from options.", 'qtranslate' ), count( $new_options ) );
+    }
+
+    $options = get_option( QTX_SLUGS_LEGACY_QTS_OPTIONS_NAME );
+    if ( $options ) {
+        $new_options = [];
```

Comment:
Not clear why we want to delete existing new named options.
Shouldn't be better to merge existing new options array and renamed old options array to preserve possible new entries added between QTX update and migration?

Reply:
QTX should not alter any of the slugs created by QTS. Those belong to QTS, it's another plugin. We should see them as read-only. What happens to those are not our concern.
We need to delete existing meta with new key in case they have already been imported once. If we repeat the import multiple times, duplicates would be created.

Reply:
Another option could also be to add "Delete existing `slugs` data before import". Then do a merge in any case.

Reply:
I mean add an additional option checkbox "Delete existing slugs data before import". Then it's all up to the admin to decide what he wants. In any case we don't touch the original QTS data so it's pretty safe.

Reply:
> Perhaps the merge could still be done, if we consider it like a PHP array_merge. The new data will only replace the old data for new keys. The old keys are kept only if they don't exist in the new array.

Yes that is what I meant. I would avoid additional checkboxes, better doing it automatically.

Reply:
The new import now selects only the QTS slugs that have not been imported yet. Existing slugs (after rename) are ignored.

Added also a new option to delete before import. This gives max flexibility in case the admin wants to erase all and restart the import from scratch.

Reply:
Done. It's a bit more tricky than it seems but it should work.
Existing meta_key are preserved unless deletion is asked explicitly. If not, a conservative merge is done:
- for new types, all slugs are imported
- for existing types, slugs are merged, existing slugs are preserved (no overwrite).

Reply:
Need to re-open this. Problem: the QTX options are saved before the QTS import is run. So new keys are created and stored in new options before the import happens. As the import is conservative preserves existing values it doesn't do anything. Need more checks on values and overwrite empty slugs strings.